packages/wrangler/src/api/startDevWorker/DevEnv.ts

```diff
@@ -181,11 +193,14 @@ function createWorkerObject(devEnv: DevEnv): Worker {
 				this.config.name,
 				"Worker name must be defined to use `Worker.scheduled()`"
 			);
-			const { proxyWorker } = await devEnv.proxy.ready.promise;
+			const ev = await devEnv.proxy.ready.promise;
+			assert(ev);
+			const { proxyWorker } = ev;
 			const w = await proxyWorker.getWorker(this.config.name);
 			return w.scheduled(...args);
 		},
 		async dispose() {
+			await devEnv.proxy.ready.promise;
 			await devEnv.teardown();
```

Comment:
What about just waiting for the `ready` promise to be settled (either resolved or rejected)?
As this stands, if the `ready` promise rejects this will throw an error and the `tearDown()` below will never get called.

```suggestion
			await devEnv.proxy.ready.promise.finally(devEnv.teardown);
```

If we do this then I think you don't need to change the type of this promise to include `null`.
Instead you can just reject the promise instead of resolving to `null`.

Reply:
Good call 🙂

I've applied the change, please have another look 🙏